specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/CommonDefinitions.json

```diff
@@ -3001,6 +3001,10 @@
               "readOnly": true,
               "example": "00000000-0000-0000-0000-000000000000"
             },
+            "publicNetworkAccess": {
+              "description": "Property to allow or block all public traffic.",
+              "type": "string"
+            },
```

Comment:
Should it be an enum?

Reply:
I agree enum would make more sense but the string property is designed by a vanguard requirement across services in Azure and we are just one of the services to implement it to meet the requirement so we need to make it a string as well to keep the consistency across services